### DISCUSSION TITLE
Exporting Custom Donor Metadata

### DISCUSSION THREAD

**Discussion by HowdyMcGee** [impress-org/givewp#7398] [Developer Q&A]
Hello,

It seems like GiveWP now uses a different export Class for Donors. I used to be able to use the `give_export_donors_get_default_columns` hook to add in labels, and the `give_export_get_data_donors` hook to add in values. It doesn't look like these hooks fire anymore. What's the best way to add in custom Donor Metadata to the export file?

**Comment by JoshuaHungDinh** [comment]
Hey @HowdyMcGee, thanks for raising this to our attention. I briefly tested these filters out and we may need to include the column filter in the DonorExport class. I'll bring this up to the team. Let's keep this discussion open for now and I'll provide you with an update through this channel.

**Comment by HowdyMcGee** [comment]
Is there anything I can do to ensure that this doesn't get lost in the dev process? Unfortunately, my client cannot export the custom metadata I've deved without these hooks.

**Reply by JoshuaHungDinh** [reply]
Hey @HowdyMcGee, just wanted to give you a quick update. We've implemented a fix for the issue we discussed earlier, and it's currently in the review process. This should address the problem with the custom metadata export hooks.

**Comment by HowdyMcGee** [comment]
Hello! I was able to get this working, but I am having trouble getting the column keys to populate.

For example, I can add the selectable checkboxes via the `give_export_donors_get_default_columns` hook, and they appear fine, but in the `give_export_get_data_donors` hook the keys aren't populated as empty strings in the data array, they're just not there. So in the actual data hook I don't know which keys were selected.

Any ideas on how I can ensure that my custom key is either selected or set while I'm in the data hook callback?

**Comment by kjohnson** [comment]
@HowdyMcGee after taking another look at this I see where the confusion is. I've put together some notes for the team and we are scoping the work to restore custom fields in the donor export.

**Comment by kjohnson** [comment]
@HowdyMcGee I have a pull request that addresses this issue, see https://github.com/impress-org/givewp/pull/7682
